packages/common/src/adapters/user.ts

```diff
@@ -22,53 +22,56 @@ import { playlistLibraryFromSDK } from './playlistLibrary'
 export const userMetadataFromSDK = (
   input: full.UserFull
 ): UserMetadata | undefined => {
-  const user = snakecaseKeys(input)
-  const decodedUserId = decodeHashId(user.id)
+  const decodedUserId = decodeHashId(input.id)
   if (!decodedUserId) {
     return undefined
   }
 
   const newUser: UserMetadata = {
     // Fields from API that are omitted in this model
-    ...omit(user, ['id', 'cover_photo_legacy', 'profile_picture_legacy']),
+    ...omit(snakecaseKeys(input), [
+      'id',
+      'cover_photo_legacy',
+      'profile_picture_legacy'
+    ]),
 
     // Conversions
-    artist_pick_track_id: user.artist_pick_track_id
-      ? decodeHashId(user.artist_pick_track_id)
+    artist_pick_track_id: input.artistPickTrackId
```

Comment:
Nice, i like that it's using `input` directly instead of after the snake_case conversion like before

Reply:
This actually turns out to be a good idea for another very important reason: `snakecaseKeys` is a deep operation by default and that messes up the logic of nested adapters if they are expecting a `full.*` SDK object with camelCase keys! 😬